internal/api/projects.go

```diff
@@ -184,7 +184,10 @@ func (p *v1Provider) PutProjectMaxQuota(w http.ResponseWriter, r *http.Request)
 	httpapi.IdentifyEndpoint(r, "/v1/domains/:id/projects/:id/max-quota")
 	requestTime := p.timeNow()
 	token := p.CheckToken(r)
-	if !token.Require(w, "project:edit_max_quota") {
+	projectAccess := token.Check("project:edit_max_quota")
+	domainAccess := token.Check("domain:edit_max_quota")
+	if !projectAccess && !domainAccess {
+		http.Error(w, "Forbidden", http.StatusForbidden)
```

Comment:
I have two notes here.

1. The pattern for policy rule names is that the first part identifies the object whose attributes are being changed. Since these endpoints update project quota constraints, all policies shall be `project:...` instead of `domain:...`. (Compare to the state before, where `project:edit_max_quota` was only accessible to domain admins.)
2. For editing `max_quota_from_local_admin`, I would like to avoid policy rule sprawl and reuse `project:edit` (which used to refer to editing of quotas, and currently refers to editing of commitments).

All in all, while also noting that domain admins are also project admins by inheritance:

```suggestion
	if !token.Require("project:edit") {
		return
	}
	domainAccess := token.Check("project:edit_as_outside_admin")
```